### ISSUE TITLE
targetHost 'localhost' is not an ADC

### ISSUE THREAD

**Issue by codygreen** [F5Networks/f5-appsvcs-extension#142]
##### ISSUE TYPE

 - Bug Report

##### AS3 BUILD/ VERSION

```
3.5.1-5
```

##### BIGIP VERSION

```
Sys::Version
Main Package
  Product     BIG-IP
  Version     13.1.1.4
  Build       0.0.4
  Edition     Point Release 4
  Date        Wed Jan 16 18:04:36 PST 2019
```

##### SUMMARY

```
Declaration returns a 404 with a message: "targetHost 'localhost' is not an ADC"
The declaration does not use the targetHost attribute
```

##### STEPS TO REPRODUCE

```
sensitive information removed and replaced with xxxx
{
    "class": "AS3",
    "action": "deploy",
    "persist": true,
    "declaration": {
        "class": "ADC",
        "schemaVersion": "3.0.0",
        "id": "example-declaration-01",
        "label": "Sample 1",
        "remark": "Simple HTTP application with round robin pool",
        "www": {
            "class": "Tenant",
            "Application_1": {
                "class": "Application",
                "template": "http",
                "serviceMain": {
                    "class": "Service_HTTP",
                    "virtualAddresses": [
                        "10.90.2.4"
                    ],
                    "pool": "web_pool",
                    "policyWAF": {
                        "bigip": "/Common/blocking.misc.prod"
                    }
                },
                "web_pool": {
                    "class": "Pool",
                    "monitors": [
                        "http"
                    ],
                    "members": [{
                        "servicePort": 80,
                        "addressDiscovery": "azure",
                        "updateInterval": 10,
                        "tagKey": "name",
                        "tagValue": "demo-dev-app-nic",
                        "addressRealm": "private",
                        "resourceGroup": "demo-dev-rg",
                        "subscriptionId": "xxxx",
                        "directoryId": "xxxx",
                        "applicationId": "xxxx",
                        "apiAccessKey": "xxxx",
                        "credentialUpdate": false

                    }]
                }
            }
        }
    }
}
```

##### EXPECTED RESULTS

```
Successful deployment of declaration
```

##### ACTUAL RESULTS

```
On some BIG-IPs the declaration works and on others, I receive 
the error stated above.  BIG-IPs in this environment are deployed 
via Terraform and the A&O toolchain is installed using the Ansible 
role.  This issue occurs in ~30% of my deployments.
```

**Comment by codygreen** [comment]
restarting restnoded does not resolve this problem.

AS3 is installed and seems to be working:
GET https://{{F5_SERVER}}/mgmt/shared/appsvcs/info returns:
{
    "version": "3.5.1",
    "release": "5",
    "schemaCurrent": "3.5.0",
    "schemaMinimum": "3.0.0"
}

**Comment by codygreen** [comment]
f5admin@(demo)(cfg-sync Standalone)(Active)(/Common)(tmos.sys.provision)# list
sys provision afm { }
sys provision am { }
sys provision apm { }
sys provision asm {
    level nominal
}
sys provision avr { }
sys provision dos { }
sys provision fps { }
sys provision gtm { }
sys provision ilx { }
sys provision lc { }
sys provision ltm {
    level nominal
}
sys provision pem { }
sys provision swg { }
sys provision urldb { }

**Comment by markchart** [comment]
To decide if the target host (default `localhost`) is a BIG-IP, AS3 probes for a valid response to an iControl REST query for `/mgmt/tm/sys/version`.  Take a look in the logs (you may have to set log-level to debug) for more info about what is going wrong.

**Comment by dstokesf5** [comment]
This was also reported by another source, and is in our internal product backlog as AT-749.

**Comment by thomsonjames** [comment]
We're also hitting this issue in a PAYG instance in Azure that has been running for months and this just all of a sudden happened.

**Comment by dstokesf5** [comment]
This issue was resolved in the 3.16.0 release.
